### PR TITLE
[tests-only] do not try to delete already deleted groups

### PR DESCRIPTION
**Pull request by saw-jan**
## Description
Do not try to delete already deleted groups

Fixes this error in AfterScenario hooks:
```php
│
  ╳  HTTP status code 404 is not the expected value 204
  ╳  Failed asserting that 404 matches expected 204.
  │
  └─ @AfterScenario # FeatureContext::afterScenario()
```

## Related Issue
https://github.com/owncloud/ocis/issues/8864

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:
- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: tests/acceptance/features/bootstrap/Provisioning.php [hunk 1]
```diff
@@ -4247,22 +4252,24 @@ public function cleanupDatabaseGroups():void {
 		$previousServer = $this->currentServer;
 		$this->usingServer('LOCAL');
 		foreach ($this->createdGroups as $group => $groupData) {
-			if ($this->isTestingWithLdap()) {
-				$this->cleanupGroup((string)$group);
-			} else {
-				$this->graphContext->adminDeletesGroupWithGroupId(
-					$groupData['id']
-				);
+			if ($groupData["possibleToDelete"]) {
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/Provisioning.php:4255]: I don't see anywhere that sets "possibleToDelete" to `true`
So the code in this block will not be executed?

Comment (saw-jan) [reply tests/acceptance/features/bootstrap/Provisioning.php:4255]: These properties will be set `true` initially while adding the created group by method `addGroupToCreatedGroupsList`
https://github.com/owncloud/ocis/blob/65213abcb4141bb36f9d5cd6de2e6f5b43016fab/tests/acceptance/features/bootstrap/Provisioning.php#L2592-L2597

Comment (phil-davis) [reply tests/acceptance/features/bootstrap/Provisioning.php:4255]: OK, that is existing code.

**Comment by saw-jan** [comment]
Still some errors left

File: tests/acceptance/features/bootstrap/SharingNgContext.php [hunk 2]
```diff
@@ -193,10 +193,16 @@ public function sendShareInvitation(string $user, TableNode $table, string $file
...
-			foreach ($sharees as $sharee) {
+			foreach ($sharees as $index => $sharee) {
+				$shareType = $shareTypes[$index];
+				$shareeId = "";
+				if ($shareType === "user") {
+					$shareeId = $this->featureContext->getAttributeOfCreatedUser($sharee, 'id');
+				} elseif ($shareType === "group") {
+					$shareeId = $this->featureContext->getAttributeOfCreatedGroup($sharee, 'id');
+				}
 				// for non-exiting group or user, generate random id
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/SharingNgContext.php:204]:
```suggestion
				// for non-existing group or user, generate random id
```
some day this typo could be fixed, but not related to this PR :)

File: tests/acceptance/features/bootstrap/Provisioning.php [hunk 3]
```diff
@@ -4247,22 +4224,24 @@ public function cleanupDatabaseGroups():void {
...
+				if ($this->isTestingWithLdap()) {
+					$this->cleanupGroup((string)$group);
+				} else {
+					$response = $this->graphContext->deleteGroupWithId($groupData['id']);
+					$this->theHTTPStatusCodeShouldBe(204, "", $response);
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/Provisioning.php:4232]: I wonder if we want to cause `cleanupDatabaseGroups` to fail if the group delete gets a 404 (if the group already does not exist).
But I suppose it is good to know if this happens - it means that some test code has created a group that didn't get put into the `createdGroups` array.

**Review by phil-davis — APPROVED** [review]

**Pull request merged**